openwisp_utils/measurements/tasks.py

```diff
@@ -0,0 +1,55 @@
+import logging
+
+from celery import shared_task
+from openwisp_utils.admin_theme.system_info import (
+    get_enabled_openwisp_modules,
+    get_openwisp_version,
+    get_os_details,
+)
+
+from ..utils import retryable_request
+from .models import OpenwispVersion
+from .utils import _get_events, get_openwisp_module_events, get_os_detail_events
+
+CLEAN_INSIGHTS_URL = 'https://analytics.openwisp.io/cleaninsights.php'
+MAX_TRIES = 3
+
+logger = logging.getLogger(__name__)
+
+
+def post_clean_insights_events(events):
+    try:
+        response = retryable_request(
+            'post',
+            url=CLEAN_INSIGHTS_URL,
+            json={
+                'idsite': 5,
+                'events': events,
+            },
+        )
+        assert response.status_code == 204
+    except Exception as error:
```

Comment:
are you absolutely sure this code works also when the server does not reply at all?

Reply:
I have tested the following scenarios: 

1. Trying to establish connection to the port where no application is listening - Check the [command output on Hastebin](https://hastebin.skyra.pw/ivuniyihat.py)
2. When the server time outs (hangs) after establishing the connection - Check the [command output on Hastebin](https://hastebin.skyra.pw/jeyowecehe.py)

I have added a test case for this https://github.com/openwisp/openwisp-utils/pull/365/files#diff-118b67d11be9c05a77bccfedd814f41974ca3fb2bfd21eb010350faf22d59d3bR180-R196